torch/distributed/distributed_c10d.py

```diff
@@ -1297,6 +1298,65 @@ def get_node_local_rank(fallback_rank: Optional[int] = None) -> int:
     )
 
 
+def _get_backends_for_timeout_set(group: ProcessGroup) -> Set[c10d_Backend]:
+    """
+    Return the all possible backends for a given process group when we
+    need to set the timeout for a ProcessGroup.
+
+    Args:
+        group (ProcessGroup): The process group to be set timeout.
+
+    Returns:
+        set[c10d_Backend]: A set of backends which we can set timeout for.
+    """
+    assert isinstance(group, ProcessGroup)
+    devices = group._device_types
+    backends = set()
+    if torch.device("cpu") in devices and is_gloo_available():
```

Comment:
kind of confused about all the if-statement checks. Is there a downside in simplifying all the logic it to just:

```
for device in devices:
    backends.add(group._get_backend(device))
```

Reply:
Oh because we only have pybind API defined for NCCL and Gloo backend, unfortunately. Or do we want to actually enable them for all backend?